lib/jellyfish/config_reader.ex

```diff
@@ -128,12 +103,121 @@ defmodule Jellyfish.ConfigReader do
     end
   end
 
-  defp parse_node_name(node_name), do: String.to_atom(node_name)
+  def read_dist_config() do
+    dist_enabled? = read_boolean("JF_DIST_ENABLED")
+    dist_strategy = System.get_env("JF_DIST_STRATEGY_NAME")
+    node_name_value = System.get_env("JF_DIST_NODE_NAME")
+    cookie_value = System.get_env("JF_DIST_COOKIE", "jellyfish_cookie")
+
+    cond do
+      is_nil(dist_enabled?) or not dist_enabled? ->
+        [enabled: false, strategy: nil, node_name: nil, cookie: nil, strategy_config: nil]
+
+      dist_strategy == "NODES_LIST" or is_nil(dist_strategy) ->
+        nodes_value = System.get_env("JF_DIST_NODES", "")
+
+        unless node_name_value do
+          raise "JF_DIST_ENABLED has been set but JF_DIST_NODE_NAME remains unset."
+        end
+
+        node_name = parse_node_name(node_name_value)
+        cookie = parse_cookie(cookie_value)
+        nodes = parse_nodes(nodes_value)
+
+        if nodes == [] do
+          Logger.warning("""
+          JF_DIST_ENABLED has been set but JF_DIST_NODES remains unset.
+          Make sure that at least one of your Jellyfish instances
+          has JF_DIST_NODES set.
+          """)
+        end
+
+        [
+          enabled: true,
+          strategy: Cluster.Strategy.Epmd,
+          node_name: node_name,
+          cookie: cookie,
+          strategy_config: [hosts: nodes]
+        ]
+
+      dist_strategy == "DNS" ->
+        do_read_dns_config(dist_strategy, node_name_value, cookie_value)
```

Comment:
Please inline this function

Reply:
The problem is that the credo suggests that this function is too complex.